packages/editor/src/client/blocks/richTextBlock.tsx

```diff
@@ -431,6 +486,139 @@ function RemoveLinkFormatButton() {
   )
 }
 
+export interface TableButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  readonly icon: IconNames | SVGIcon
+  readonly iconActive?: IconNames | SVGIcon
+  readonly active?: boolean
+}
+
+export const BaseTableButton = forwardRef<HTMLButtonElement, TableButtonProps>(
+  function BaseTableButton({icon, iconActive, active, ...props}, ref) {
+    icon = iconActive && active ? iconActive : icon
+    return (
+      <button
+        style={{
+          border: active ? 'blue 1px solid' : '',
+          fontSize: 16,
+
+          cursor: 'pointer',
+          borderRadius: 3,
+          backgroundColor: 'transparent',
+
+          padding: 2
+        }}
+        ref={ref}
+        {...props}>
+        <Icon icon={icon} element={icon} />
+      </button>
+    )
+  }
+)
+
+function TableButton({icon, iconActive}: TableButtonProps) {
+  const editor = useSlate()
+  const [nrows, setNrows] = useState(2)
+  const [ncols, setNcols] = useState(1)
+
+  const [isTableSpecsOpen, setIsTableSpecsOpen] = useState(false)
+
+  const triggerRef = useRef<any>(null)
+
+  const {t} = useTranslation()
+
+  const open = () => {
+    if (triggerRef.current && !isFormatActive(editor, BlockFormat.Table)) {
+      triggerRef.current!.open()
+      // TODO rather be done with forwardRef ?
+      setIsTableSpecsOpen(true)
+    }
+  }
+  const close = () => {
+    if (triggerRef.current) {
+      triggerRef.current!.close()
+      setIsTableSpecsOpen(false)
+    }
+  }
+
+  const tableSpecs = (
+    <Popover>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-between',
+          height: '130px'
+        }}>
+        <SetRowColNumbers key={'row'} label="rows" num={nrows} setNumber={setNrows} />
+        <SetRowColNumbers key={'col'} label="columns" num={ncols} setNumber={setNcols} />
+        <Button
+          disabled={isFormatActive(editor, BlockFormat.Table)}
+          onClick={() => {
+            Transforms.insertNodes(editor, emptyCellsTable(nrows, ncols))
+            close()
+          }}>
+          {t('insert table')}
```

Comment:
We have chosen to use a specific way of formatting translations. Compare with `locales/en.json`: it would be something like `{t('blocks.richTextTable.insertTable')}`

Reply:
👍🏾

Reply:
Ok, thanks, will correct.